[WIP] feat: Smallest functional test for chainPR

yoshwata opened the pull request:
## Context

<!-- Why do we need this PR? What was the reason that led you to make this change? -->
This PR adds smallest functional tests to keep the quality of chainPR

## Objective

<!-- What does this PR fix? What intentional changes will this PR make? -->
This PR adds a new test for chainPR. Details can be found in the feature file.

However, it doesn't work because it was intentionally disabled.

In order to run it, [testOrg](https://github.com/screwdriver-cd-test) needs a new repository and screwdriver.yaml. The repository name supposed to be `functional-chainPR`

After being merged, authorized members should add the repository and screwdriver.yaml and remove [`@ignore`](https://github.com/screwdriver-cd/screwdriver/compare/master...sonic-screwdriver-cd:add-chainpr-test?expand=1#diff-4e0f06952f23b6671797dca68cb8fb191d0b29063f14f140a5805cb82f4e87f6R144).
Also, you can add the repository first. In that case, `ignore` is removed in this PR

The minimum chainPR configuration is as follows, and it's also supposed to be yaml in the `functional-chainPR` repo.
```
annotations:
  screwdriver.cd/chainPR: true

shared:
    image: node:14
jobs:
    SIMPLE:
        steps:
            - echo: echo test
        requires: [~pr, ~commit]
    AFTER-SIMPLE:
        steps:
            - echo: echo test
        requires: [SIMPLE]
```

The workflow is illustrated below.
<img width="1121" alt="functional-chainPR-graph" src="https://user-images.githubusercontent.com/32473622/127270620-17521b33-bccc-4cf2-9970-8badea249370.png">


I have confirmed that workflow tests pass in my local environment.

## References

<!-- Links or resources that help clarify and support your intentions (e.g., Github issue) -->

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

yoshwata commented:
I will fix lint

wahapo commented on features/workflow.feature:
```diff
@@ -140,3 +140,15 @@ Feature: Workflow
+    Scenario: chainPR
+        Given an existing pipeline on "master" branch with the workflow jobs:
+            | job       | requires  |
+            | SIMPLE    | ~commit   |
```
```suggestion
            | SIMPLE    | ~pr       |
```

yoshwata replied:
Thank you. Fixed with table misalignment.

tkyi commented on features/workflow.feature:
```diff
@@ -140,3 +140,15 @@ Feature: Workflow
+
+    @ignore @workflow-chainPR
+    Scenario: chainPR
+        Given an existing pipeline on "master" branch with the workflow jobs:
```
should we change this to `"main" branch`? I know GitHub defaults to `main` branch moving forward

I'm open to changing the repo default branch to "master" if you think that makes more sense

tkyi reviewed (commented):
Created test repo with yaml config: https://github.com/screwdriver-cd-test/functional-chainPR
Also invited you as Owner for screwdriver-cd-test repo

Just need to figure out default branch name

yoshwata commented:
@tkyi Since some APIs are based on master, this time I will proceed with the default as master. I've changed default branch as master about `functional-chainPR` repo.

https://github.com/screwdriver-cd/screwdriver/blob/250169277acf950238b94adfa7fb7dbebb1f1304/features/support/github.js#L94

https://github.com/screwdriver-cd/screwdriver/blob/250169277acf950238b94adfa7fb7dbebb1f1304/features/support/github.js#L151

Thank you for your perfect work. I've removed `@ignore`.

tk3fftk commented:
related: https://github.com/screwdriver-cd/screwdriver/issues/1393

tkyi commented on features/workflow.feature:
```diff
@@ -92,6 +92,18 @@ Feature: Workflow
 
+    @workflow-chainPR
+    Scenario: chainPR
+        Given an existing pipeline on "main" branch with the workflow jobs:
```
```suggestion
        Given an existing pipeline on "master" branch with the workflow jobs:
```

tk3fftk commented on features/support/sdapi.js:
```diff
@@ -18,6 +18,31 @@ function promiseToWait(timeToWait) {
+ *                                              empty array is returned
+ */
+function findJobs(config) {
+    const { instance } = config;
+    const { pipelineId } = config;
```
small refactor
```suggestion
    const { instance, pipelineId } = config;
```

tk3fftk reviewed (approved):
:+1:

yoshwata commented:
We have found that this test has a high probability of failure in our environment. I will investigate.

yoshwata commented:
Closed for rethinking.
